gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,588 @@
+    private final String body;
+
+    Response(int code, String body) {
+      this.code = code;
+      this.body = body;
+    }
+
+    int code() {
+      return code;
+    }
+
+    String body() {
+      return body;
+    }
+  }
+
+  enum Error {
+    ALREADY_EXISTS(
+        409, "global", "Requested entity already exists.", "alreadyExists", "ALREADY_EXISTS"),
+    PERMISSION_DENIED(
+        403, "global", "The caller does not have permission.", "forbidden", "PERMISSION_DENIED"),
+    FAILED_PRECONDITION( // change this error code to 412 when #440 is fixed
+        400, "global", "Precondition check failed.", "failedPrecondition", "FAILED_PRECONDITION"),
+    INVALID_ARGUMENT( // change this error code to 412 when #440 is fixed
+        400, "global", "Request contains an invalid argument.", "badRequest",
+        "INVALID_ARGUMENT");
+
+    private final Response response;
+
+    Error(int code, String domain, String message, String reason, String status) {
+      this.response = new Response(code, toJson(code, domain, message, reason, status));
+    }
+
+    private static String toJson(
+        int code, String domain, String message, String reason, String status) {
+      Map<String, Object> args = new HashMap<>();
+      Map<String, Object> nestedArgs = new HashMap<>();
+      nestedArgs.put("domain", domain);
+      nestedArgs.put("message", message);
+      nestedArgs.put("reason", reason);
+      List<Map<String, Object>> errors = ImmutableList.of(nestedArgs);
+      args.put("errors", errors);
+      args.put("code", code);
+      args.put("message", message);
+      args.put("status", status);
+      try {
+        return jsonFactory.toString(ImmutableMap.of("error", args));
+      } catch (IOException e) {
+        throw new RuntimeException("Error when generating JSON error response.");
+      }
+    }
+  }
+
+  private class RequestHandler implements HttpHandler {
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+      // see https://cloud.google.com/resource-manager/reference/rest/
+      String path = exchange.getRequestURI().getPath();
+      String requestMethod = exchange.getRequestMethod();
+      Response response = null;
+      if (requestMethod.equals("POST") && path.startsWith("/v1beta1/projects")) {
+        if (path.contains("undelete")) {
+          response = undelete(projectIdFromURI(path));
+        } else {
+          String requestBody =
+              decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+          response = create(jsonFactory.fromString(requestBody, Project.class));
+        }
+      } else if (requestMethod.equals("DELETE")) {
+        response = delete(projectIdFromURI(path));
+      } else if (requestMethod.equals("GET") && path.startsWith("/v1beta1/projects/")) {
+        response = get(projectIdFromURI(path), parseFields(exchange.getRequestURI().getQuery()));
+      } else if (requestMethod.equals("GET")) {
```

Comment:
This could have been combined with the prior GET (and inside check for path match).